server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1092,4 +1155,61 @@ protected void parseCreateField(ParseContext context) throws IOException {
     public FieldMapper.Builder getMergeBuilder() {
         return new Builder(simpleName(), type, ignoreMalformedByDefault, coerceByDefault).init(this);
     }
+
+    @Override
+    public void doPostParse(ParseContext context, IndexTimeScriptParams params) {
+        if (this.script != null) {
+            script.execute(params, v -> indexValue(context, v));
+        }
+    }
+
+    public static final ScriptContext<ScriptFactory> SCRIPT_CONTEXT = new ScriptContext<>(
+        "number_script", ScriptFactory.class
+    );
```

Comment:
It may be possible to combine this with the equivalent runtime fields implementations by sharing a common interface; I will investigate this next.

Reply:
yes, I would hope that we can share the script context entirely, once it is registered as part of the builtin contexts.